### DISCUSSION TITLE
Low performance scores in Lighthouse and PageSpeed testing

### DISCUSSION THREAD

**Discussion by ywtstewart** [vercel/next.js#13674] [Help]
Hello fello devs, 

I am just wondering if there's anyone who has encountered low mobile performance scores (in Lighthouse: 50 and Pagespeed: 30) and that have been able to increase the score for that specific metric.  

My stack:

- Next.js 9.4.4 (custom express server)
- Apollo Client (using strategy from with-apollo example)
- Styled components 

The issues I'm seeing are: 

- Minimize main-thread work 
- Reduce the impact of third-party code (imported with Google TagManager
- Reduce JavaScript execution time
- Avoid enormous network payloads

**Comment by hk86** [comment]
I've the same issue. When I check with https://www.webpagetest.org I get completely different metrics.  I.e. First Contentful Paint is saying 2.5s. If I check the screenshots taken, I see that on the first image the site is already fully loaded. I have no clue what is the reason behind this. I deployed on Vercel.

**Comment by prKassad** [comment]
I am also saddened by results in Lighthouse. Even a simple application (like learn-starter) has FCP and TTI more than 2 seconds. However, in reality, the page renders faster than these metrics.

**Comment by hk86** [comment]
Is it an issue with Lighthouse or Next.js?

**Reply by timneutkens** [reply]
See https://github.com/vercel/next.js/discussions/13682

**Reply by hk86** [reply]
This issue does not explain the bad metrics (i.e. FCP more then 2s).

**Reply by timneutkens** [reply]
Hard to give you any guidance as this is not the case for all Next.js applications, it's definitely related to your application and we'd need a full reproduction including code to investigate.

**Comment by JOEMN96** [comment]
![next](https://user-images.githubusercontent.com/56827756/106412918-cd641300-646e-11eb-8bf4-d5a277a5917a.PNG)

any update on this ?

**Reply by benweier** [reply]
@JOEMN96 have you run the lighthouse test against a production build/environment? The dev server injects a bunch of extra code that tanks performance scores.
